core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -635,6 +635,30 @@ class ZkUtils(val zkClient: ZkClient,
     zkClient.deleteRecursive(path)
   }
 
+  def subscribeDataChanges(path: String, listener: IZkDataListener): Unit = {
+    zkClient.subscribeDataChanges(path, listener)
+  }
+
+  def unsubscribeDataChanges(path: String, dataListener: IZkDataListener): Unit = {
+    zkClient.unsubscribeDataChanges(path, dataListener)
+  }
+
+  def subscribeStateChanges(listener: IZkStateListener): Unit = {
+    zkClient.subscribeStateChanges(listener)
+  }
+
+  def subscribeChildChanges(path: String, listener: IZkChildListener): Seq[String] = {
+    zkClient.subscribeChildChanges(path, listener).asScala
```

Comment:
It looks like `subscribeChildChanges` returns `null` if `path` doesn't exist. I changed the method to return `Option[Seq[String]]`.